todoApp/src/App.tsx

```diff
@@ -23,29 +24,41 @@ const App = () => {
     setTodoTasks(todoArray);
   };
 
-  const curTime = new Date();
-  todoTasks = todoTasks.filter((todo: TodoListInterface) => {
-    let curDate = `${curTime.getDate()}/${
-      curTime.getMonth() + 1
-    }/${curTime.getFullYear()}`;
-    return curDate === todo.date;
-  });
-  localStorage.setItem("todo", JSON.stringify(todoTasks));
+  let curDate = `${moment().format("D")}/${moment().format(
+    "M"
+  )}/${moment().format("YYYY")}`;
+  useEffect(() => {
+    setTodoTasks((prev) =>
+      prev.filter((todo: TodoListInterface) => {
+        return curDate === todo.date;
+      })
+    );
+  }, [curDate]);
+
+  const checkTodoHandler = (id: string) => {
+    const getIndex = todoTasks.findIndex((task) => task.id === id);
+    const updateTasks = [...todoTasks];
+    updateTasks[getIndex].isCompleted = !updateTasks[getIndex].isCompleted;
+    setTodoTasks(updateTasks);
```

Comment:
This approach is correct but for such use case map function with a condition is recommended.